src/gas/gas-util.test.ts

```diff
@@ -50,7 +54,75 @@ const mockEIP1559ApiResponses = [
   },
 ];
 
-describe('gas utils', () => {
+describe('gas-util', () => {
+  describe('normalizeGWEIDecimalNumbers', () => {
+    describe('given a number', () => {
+      it('should return a whole number as a string if no normalization is needed', () => {
+        expect(normalizeGWEIDecimalNumbers(1)).toBe('1');
+        expect(normalizeGWEIDecimalNumbers(123)).toBe('123');
+        expect(normalizeGWEIDecimalNumbers(101)).toBe('101');
+        expect(normalizeGWEIDecimalNumbers(1234)).toBe('1234');
+        expect(normalizeGWEIDecimalNumbers(1000)).toBe('1000');
+      });
+
+      it('should return a decimal number greater than 1 as a string if no normalization is needed', () => {
+        expect(normalizeGWEIDecimalNumbers(1.1)).toBe('1.1');
+        expect(normalizeGWEIDecimalNumbers(123.01)).toBe('123.01');
+        expect(normalizeGWEIDecimalNumbers(101.001)).toBe('101.001');
+        expect(normalizeGWEIDecimalNumbers(100.001)).toBe('100.001');
+        expect(normalizeGWEIDecimalNumbers(1234.567)).toBe('1234.567');
+      });
+
+      it('should return a decimal number less than 1 as a string if no normalization is needed', () => {
+        expect(normalizeGWEIDecimalNumbers(0.1)).toBe('0.1');
+        expect(normalizeGWEIDecimalNumbers(0.01)).toBe('0.01');
+        expect(normalizeGWEIDecimalNumbers(0.001)).toBe('0.001');
+        expect(normalizeGWEIDecimalNumbers(0.567)).toBe('0.567');
+      });
+
+      it('should round a decimal number to 9 places before converting to a string', () => {
```

Comment:
How does this function handle numbers that exceed the limits of precision for JavaScript floats? Also for MAX_INT, MIN_INT. Negative numbers in general really